### PR TITLE
fby3.5: hd: Support second source sensor

### PR DESCRIPTION
**Pull request by Yi-Shum**
Summary:
- Support temperature sensor G788P87U.
- Support VR sensor xdpe19283b.
- Check HSC sensor type by GPIO HSC_TYPE_0/1.
- Check VR and temperature sensor type by board revision.

Dependency: #462 

Test Plan:
- Build Code: PASS.

Log:
Second source board
```
root@bmc-oob:~# sensor-util slot1
...
HSC Temp                     (0xE) :   48.12 C     | (ok)
CPU0 VR Temp                 (0xF) :   57.00 C     | (ok)
SOC VR Temp                  (0x10) :   56.00 C     | (ok)
CPU1 VR Temp                 (0x11) :   38.00 C     | (ok)
PVDDIO VR Temp               (0x12) :   38.00 C     | (ok)
PVDD11 VR Temp               (0x13) :   36.00 C     | (ok)
P12V_STBY Vol                (0x14) :   12.11 Volts | (ok)
PVDD18_S5 Vol                (0x15) :    1.83 Volts | (ok)
P3V3_STBY Vol                (0x16) :    3.34 Volts | (ok)
PVDD11_S3 Vol                (0x17) :    1.11 Volts | (ok)
P3V_BAT Vol                  (0x18) :    3.19 Volts | (ok)
PVDD33_S5 Vol                (0x19) :    3.34 Volts | (ok)
P5V_STBY Vol                 (0x1A) :    5.01 Volts | (ok)
P12V_MEM_1 Vol               (0x1B) :   12.11 Volts | (ok)
P12V_MEM_0 Vol               (0x1C) :   12.13 Volts | (ok)
P1V2_STBY Vol                (0x1D) :    1.21 Volts | (ok)
P3V3_M2 Vol                  (0x1E) :    3.33 Volts | (ok)
P1V8_STBY Vol                (0x1F) :    1.81 Volts | (ok)
HSC Input Vol                (0x20) :   12.05 Volts | (ok)
CPU0 VR Vol                  (0x21) :    0.92 Volts | (ok)
SOC VR Vol                   (0x22) :    1.19 Volts | (ok)
CPU1 VR Vol                  (0x23) :    0.92 Volts | (ok)
PVDDIO VR Vol                (0x24) :    1.10 Volts | (ok)
PVDD11 VR Vol                (0x25) :    1.10 Volts | (ok)
HSC Output Cur               (0x26) :   13.26 Amps  | (ok)
CPU0 VR Cur                  (0x27) :    8.56 Amps  | (ok)
SOC VR Cur                   (0x28) :   60.31 Amps  | (ok)
CPU1 VR Cur                  (0x29) :   13.69 Amps  | (ok)
PVDDIO VR Cur                (0x2A) :   29.00 Amps  | (ok)
PVDD11 VR Cur                (0x2B) :    2.00 Amps  | (ok)
HSC Input Pwr                (0x2C) :  159.84 Watts | (ok)
CPU0 VR Pwr                  (0x2D) :    7.88 Watts | (ok)
SOC VR Pwr                   (0x2E) :   72.12 Watts | (ok)
CPU1 VR Pwr                  (0x2F) :   12.38 Watts | (ok)
PVDDIO VR Pwr                (0x30) :   31.88 Watts | (ok)
PVDD11 VR Pwr                (0x31) :    2.25 Watts | (ok)
...
```

### CODE REVIEW

**Comment by Yi-Shum** [comment]
Hi @garnermic ,
I have fixed the conflict and do the force update.
Please help to review again, thank you.

File: common/shell/shell_platform.h [hunk 1]
```diff
@@ -67,16 +67,18 @@ const char *const sensor_type_name[] = {
...
-				ltc4282) sensor_name_to_num(fan) sensor_name_to_num(tmp431)
-				sensor_name_to_num(pmic) sensor_name_to_num(
-					ina233) sensor_name_to_num(isl69254)
-					sensor_name_to_num(max16550a) sensor_name_to_num(
-						ina230) sensor_name_to_num(xdpe12284c)
-						sensor_name_to_num(raa229621) sensor_name_to_num(
-							nct7718w) sensor_name_to_num(ltc4286)
-							sensor_name_to_num(amd_tsi)
-								sensor_name_to_num(apml_mailbox)
+			sensor_name_to_num(xdpe15284) sensor_name_to_num(ltc4282) sensor_name_to_num(
```
Comment (garnermic) [inline common/shell/shell_platform.h:70]: Please reformat this to make it easier to read.

File: meta-facebook/yv35-hd/src/platform/plat_class.c [hunk 2]
```diff
@@ -269,7 +279,34 @@ void init_platform_config()
 				printf("Unknown the 1OU card type, the voltage of ADC channel-6 is %fV\n",
 				       voltage);
 			} else if (_1ou_status.card_type == TYPE_1OU_RAINBOW_FALLS) {
```
Comment (garnermic) [inline meta-facebook/yv35-hd/src/platform/plat_class.c:281]: Please add an else statement at the end of the else if to ensure anything not handled by the if/else-if cases is caught and processed.

File: meta-facebook/yv35-hd/src/platform/plat_class.c [hunk 3]
```diff
@@ -269,7 +279,34 @@ void init_platform_config()
...
-				gpio_set(HSC_OCP_GPIO1_R, GPIO_HIGH);
+				if (hsc_module == HSC_MODULE_ADM1278) {
+					gpio_set(HSC_OCP_GPIO1_R, GPIO_HIGH);
+				} else if (hsc_module == HSC_MODULE_LTC4282) {
```
Comment (garnermic) [inline meta-facebook/yv35-hd/src/platform/plat_class.c:284]: Please add else statement to catch invalid hsc_module values.

**Review by garnermic — CHANGES_REQUESTED** [review]
Few minor changes to resolve.  Thanks!

File: common/shell/shell_platform.h (hunk 1 above)
Comment (Yi-Shum) [reply common/shell/shell_platform.h:70]: OK, I'll disable clang-format of this array.

File: meta-facebook/yv35-hd/src/platform/plat_class.c (hunk 2 above)
Comment (Yi-Shum) [reply meta-facebook/yv35-hd/src/platform/plat_class.c:281]: An else statement will be added to handle all conditions.

File: meta-facebook/yv35-hd/src/platform/plat_class.c (hunk 3 above)
Comment (Yi-Shum) [reply meta-facebook/yv35-hd/src/platform/plat_class.c:284]: An else statement will be added to handle all conditions.

**Comment by Yi-Shum** [comment]
Hi @garnermic  ,
I modified it based on the comments.
Please review again, thank you.

**Pull request closed without merging**